res/controllers/Numark-Mixtrack-Platinum-FX-scripts.js

```diff
@@ -0,0 +1,1899 @@
+MixtrackPlatinumFX.toggleFXControlSuper = false;
+
+MixtrackPlatinumFX.shifBrowseIsZoom = false;
+
+// setting this to false sets tap the file bpm, but without a way to reset its dangerous
+MixtrackPlatinumFX.tapChangesTempo = true;
+
+// pitch ranges
+// add/remove/modify steps to your liking
+// default step must be set in Mixxx settings
+// setting is stored per deck in pitchRange.currentRangeIdx
+MixtrackPlatinumFX.pitchRanges = [0.08, 0.16, 0.5];
+
+MixtrackPlatinumFX.HIGH_LIGHT = 0x7F;
+MixtrackPlatinumFX.LOW_LIGHT = 0x01;
+
+// whether the corresponding Mixxx option is enabled
+// (Settings -> Preferences -> Waveforms -> Synchronize zoom level across all waveforms)
+MixtrackPlatinumFX.waveformsSynced = true;
+
+// jogwheel
+MixtrackPlatinumFX.jogScratchSensitivity = 1024;
+MixtrackPlatinumFX.jogScratchAlpha = 1; // do NOT set to 2 or higher
+MixtrackPlatinumFX.jogScratchBeta = 1/32;
+MixtrackPlatinumFX.jogPitchSensitivity = 10;
+MixtrackPlatinumFX.jogSeekSensitivity = 10000;
+
+// blink settings
+MixtrackPlatinumFX.enableBlink = true;
+MixtrackPlatinumFX.blinkDelay = 700;
+
+// autoloop sizes, for available values see:
+// https://manual.mixxx.org/2.3/en/chapters/appendix/mixxx_controls.html#control-[ChannelN]-beatloop_X_toggle
+MixtrackPlatinumFX.autoLoopSizes = [
+    "0.0625",
+    "0.125",
+    "0.25",
+    "0.5",
+    "1",
+    "2",
+    "4",
+    "8"
+];
+
+// beatjump values, for available values see:
+// https://manual.mixxx.org/2.3/en/chapters/appendix/mixxx_controls.html#control-[ChannelN]-beatjump_X_forward
+// underscores (_) at the end are needed because numeric values (e.g. 8) have two underscores (e.g. beatjump_8_forward),
+// but "beatjump_forward"/"beatjump_backward" have only one underscore
+MixtrackPlatinumFX.beatJumpValues = [
+    "0.0625_",
+    "0.125_",
+    "0.25_",
+    "0.5_",
+    "1_",
+    "2_",
+    "", // "beatjump_forward"/"beatjump_backward" - jump by the value selected in Mixxx GUI (4 by default)
+    "8_"
+];
+
+// dim all lights when inactive instead of turning them off
+components.Button.prototype.off = MixtrackPlatinumFX.LOW_LIGHT;
+
+// pad modes control codes
+MixtrackPlatinumFX.PadModeControls = {
+    HOTCUE: 0x00,
+    AUTOLOOP: 0x0D,
+    FADERCUTS: 0x07,
+    SAMPLE1: 0x0B,
+    BEATJUMP: 0x01, // DUMMY not used by controller
+    SAMPLE2: 0x0F,
+    AUTOLOOP2: 0x0E, // DUMMY not used by controller
+    KEYPLAY: 0x0C, // DUMMY not used by controller
+    HOTCUE2: 0x02,
+    FADERCUTS2: 0x03, // DUMMY not used by controller
+    FADERCUTS3: 0x04, // DUMMY not used by controller
+    AUTOLOOP3: 0x05, // DUMMY not used by controller
+};
+
+// enables 4 bottom pads "fader cuts" for 8
+MixtrackPlatinumFX.faderCutSysex8 = [0xF0, 0x00, 0x20, 0x7F, 0x03, 0xF7];
+// enables only 4 top pads "fader cuts"
+MixtrackPlatinumFX.faderCutSysex4 = [0xF0, 0x00, 0x20, 0x7F, 0x13, 0xF7];
+
+// state variable, don't touch
+MixtrackPlatinumFX.shifted = false;
+
+MixtrackPlatinumFX.initComplete=false;
```

Comment:
would be nice to remove this. its a rather ugly workaround and fixing the root problem would be better.

Reply:
I will need more research or help on this.

Reply:
I'll try to find some time to look into it. I'm just saying its a code smell for now.